### DISCUSSION TITLE
php driver for edgedb?

### DISCUSSION THREAD

**Discussion by juliandarley** [geldata/gel#3488] [Ideas]
edgedb looks very interesting, however much of our previous work uses php and mysql. is there a php driver for edgedb or can it be used with the existing PDO postgres driver?

thx, julian

**Comment by elprans** [comment]
There's no official PHP driver yet, but looks like someone had built [some version of a binding](https://twitter.com/guikingone/status/1367028574524493831?s=21).

**Reply by TeddyBear06** [reply]
@Guikingone Do you have any plan to release it (or it's perhaps already done but I didn't find it)?

**Comment by TeddyBear06** [comment]
Hi,

I've created a naive unofficial PHP wrapper for EdgeDB using its HTTP protocol.

Still in early stages but with the help of the community we can polish it.

It is available here on GitHub : https://github.com/TeddyBear06/edgedb-php

@juliandarley EdgeDB is far from existing RDBMS, if I'm not mistaken, I think that you'll need queries rewrite as the paradigm is totally different.

**Reply by juliandarley** [reply]
@TeddyBear06 thanks for your reply and your repo, which we (Merlin AI) will take a look at. we don't currently intend to replace old systems with edgedb, but rather we would like to try it for new applications, we expect to have to learn new paradigms, terms, language and concepts.

**Comment by Guikingone** [comment]
Hi 👋🏻 

@TeddyBear06 It could be released, for now, it only supports HTTP / GraphQL (AFAIK, GraphQL is not supported anymore?) requests but it almost fully support the EdgeQL part (administration, data definition, query and so on).

Here's the repository: https://github.com/Guikingone/EdgeDBPHP (branch `http_client` to see the whole spectrum)

**Reply by Guikingone** [reply]
And to be fair, yes, migrating the TCP part is planned, the current EdgeDB protocol is harder to implement but it's not impossible regarding a PHP client 🙂

**Reply by TeddyBear06** [reply]
@Guikingone Thanks for sharing.

**Reply by tailhook** [reply]
Does PHP still needs C plugin to have persistent connection?

If this is the case, you might use Rust implementation for that plugin, in the same model as we use it for wasm: the part that is written in Rust will handle sockets, connection state and basic packets parsing (so ensures that even if PHP request ended unexpectedly, connection is in a consistent state), and the PHP part that deserializes the actual data into PHP-native types and have high-level operations.

**Reply by Guikingone** [reply]
I'm not sure that wrapping PHP with Rust is required here, PHP can handle sockets, connections and packets parsing, in the end, the questions is: Is there any benefits of wrapping it with Rust? 

IMHO, the answer is no but that's my POV 🙂

**Reply by juliandarley** [reply]
thanks for all of the above ideas and discussion. we (Merlin AI) have started learning Rust and WebAssembly, and hope that these will become central parts of our services. we are not sure when we will begin experiments, though we have already tried - as yet without success sadly - to get ffmpeg & ffprobe to work in wasm. this is a problem we simply have to solve for so many reasons!
